tests/std/tests/GH_004109_iter_value_t_direct_initialization/test.compile.pass.cpp

```diff
@@ -0,0 +1,700 @@
+    make_heap(nil, nil);
+    make_heap(nil, nil, comp);
+    sort_heap(nil, nil);
+    sort_heap(nil, nil, comp);
+    (void) is_heap(nil, nil);
+    (void) is_heap(nil, nil, comp);
+    (void) is_heap_until(nil, nil);
+    (void) is_heap_until(nil, nil, comp);
+
+    (void) min_element(nil, nil);
+    (void) min_element(nil, nil, comp);
+    (void) min_element(begin(arr), begin(arr) + 1);
+    (void) min_element(begin(arr), begin(arr) + 1, comp);
+    (void) max_element(nil, nil);
+    (void) max_element(nil, nil, comp);
+    (void) max_element(begin(arr), begin(arr) + 1);
+    (void) max_element(begin(arr), begin(arr) + 1, comp);
+    (void) minmax_element(nil, nil);
+    (void) minmax_element(nil, nil, comp);
+    (void) minmax_element(begin(arr), begin(arr) + 1);
+    (void) minmax_element(begin(arr), begin(arr) + 1, comp);
+    (void) lexicographical_compare(nil, nil, nil, nil);
+    (void) lexicographical_compare(begin(arr), end(arr), nil, nil);
+    (void) lexicographical_compare(nil, nil, begin(arr), end(arr));
+    (void) lexicographical_compare(nil, nil, nil, nil, comp);
+    (void) lexicographical_compare(begin(arr), end(arr), nil, nil, comp);
+    (void) lexicographical_compare(nil, nil, begin(arr), end(arr), comp);
+    next_permutation(nil, nil);
+    next_permutation(nil, nil, comp);
+    prev_permutation(nil, nil);
+    prev_permutation(nil, nil, comp);
+    (void) accumulate(nil, nil, val, binop);
+#if _HAS_CXX17
+    (void) reduce(nil, nil, val, binop);
+#endif // _HAS_CXX17
+    (void) inner_product(nil, nil, nil, val, binop, binop);
+#if _HAS_CXX17
+    (void) transform_reduce(nil, nil, nil, val, binop, binop);
+    (void) transform_reduce(nil, nil, val, binop, unop);
+#endif // _HAS_CXX17
+    partial_sum(nil, nil, nil, binop);
+#if _HAS_CXX17
+    exclusive_scan(nil, nil, nil, val, binop);
+    inclusive_scan(nil, nil, nil, binop);
+    inclusive_scan(nil, nil, nil, binop, val);
+    transform_exclusive_scan(nil, nil, nil, val, binop, unop);
+    transform_inclusive_scan(nil, nil, nil, binop, unop);
+    transform_inclusive_scan(nil, nil, nil, binop, unop, val);
+#endif // _HAS_CXX17
+    adjacent_difference(nil, nil, nil, binop);
+    iota(nil, nil, val);
+    uninitialized_copy(nil, nil, nil);
+    uninitialized_copy_n(nil, zero, nil);
+    uninitialized_fill(nil, nil, val);
+    uninitialized_fill_n(nil, zero, val);
+
+#ifdef __cpp_lib_execution
+    using namespace std::execution;
+    (void) all_of(par, nil, nil, pred);
+    (void) any_of(par, nil, nil, pred);
+    (void) none_of(par, nil, nil, pred);
+    for_each(par, nil, nil, pred);
+    for_each_n(par, nil, zero, pred);
+    (void) find(par, nil, nil, val);
+    (void) find_if(par, nil, nil, pred);
+    (void) find_if_not(par, nil, nil, pred);
+    (void) find_end(par, nil, nil, nil, nil);
+    (void) find_end(par, nil, nil, nil, nil, comp);
+    (void) find_first_of(par, nil, nil, nil, nil);
+    (void) find_first_of(par, nil, nil, nil, nil, comp);

```

Comment:
This is essentially a stylistic nitpick in this compile-only test, but these calls to parallel `find_end` and `find_first_of` are passing `comp` when these algorithms aren't taking less-than comparators. These should be given `pred2` since they're taking equality-like binary predicates (as the serial calls are properly passing).

Occurs below for parallel `unique` and `unique_copy`.